### PR TITLE
Add TryReadHumidity/TryReadTemperature

### LINKED ISSUE #1445: The islastread successful of dht22 is always false. Even if the correct temperature is read, this is always false?
The islastread successful of dht22 is always false. Even if the correct temperature is read, this is always false?

### PR DESCRIPTION
**Pull request by pgrawehr**
Fixes #1445 

Adds TryReadXXX methods to Dht sensor binding. Tested with DHT22 on RPI4.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1701)

### CODE REVIEW

File: src/devices/Dhtxx/DhtBase.cs [hunk 1]
```diff
@@ -274,6 +299,50 @@ internal virtual void ReadThroughI2c()
             }
         }
 
+        /// <summary>
+        /// Returns the current temperature
+        /// </summary>
+        /// <param name="temperature">[Out] The current temperature on success</param>
+        /// <returns>True on success, false if reading failed</returns>
+        public bool TryReadTemperature(
```
[the remark this answers is not in the document]
Comment (pgrawehr) [reply src/devices/Dhtxx/DhtBase.cs:324]: Moved attribute to new methods.

**Comment by pgrawehr** [comment]
@krwq @joperezr Updated API as discussed in call, marking the properties as obsolete. The internal organisation of these classes is a bit of a mess. I added a new `ValueArray<T>` class that should help working with small arrays without the need to copy them around or perform unnecessary allocations all the time. Useful when working with small arrays, as used in I2C or SPI transfers.

**Comment by pgrawehr** [comment]
Do not merge - There seems to be an issue that the code we use for some of the DHT types is incorrect for negative temperatures. I need to check all data sheets.

**Comment by pgrawehr** [comment]
DHT22 now verified in negative temperatures. Our formula was correct, the one in the Adafruit library is probably not, and the documentation is confusing.

**Review by Ellerbach — APPROVED** [review]
Thanks for checking everything with all sensors!

**Pull request merged**